app/src/main/java/com/github/sdp_begreen/begreen/fragments/UserPhotosViewAdapter.kt

```diff
@@ -44,41 +58,43 @@ class UserPhotosViewAdapter(
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
 
-        val photo = photos?.get(position)
+        lifecycleScope.launch {
 
-        if(isFeed) {
+            val photo = photos?.get(position)
+            val userId = photo?.takenBy ?: return@launch
 
-            // Display avatar if on feed
-            val drawable = ContextCompat.getDrawable(holder.avatarView.context, R.drawable.ic_baseline_person)
-            val defaultAvatar = drawable?.toBitmap()
-            // holder.avatarView.setImageBitmap(getFromDB(photo) ?: defaultAvatar)
-            holder.avatarView.setImageBitmap( defaultAvatar)
+            val user = db.getUser(userId)
 
-        }else{
-            // Do not display avatar if not on feed
-            holder.avatarView.visibility = View.GONE
-        }
+            if (isFeed) {
 
-        if (photo is TrashPhotoMetadata) {
-            // use the photo variable as a TrashPhotoMetadata instance
-
-            //Set default value
-            holder.titleView.text = photo?.caption ?: "No title"
-            holder.subtitleView.text = (photo?.takenOn?.toString()
-                ?: "Unknown date") + " | " + (photo?.trashCategory?.title
-                ?: "No category")
-            //holder.photoView.setImageBitmap(photo.getPhotoFromDataBase())
-            //TODO------------FOR DEMO -----------------
-            val url = URL("https://picsum.photos/400")
-            holder.photoView.setImageBitmap(
-                BitmapFactory.decodeStream(
-                    url.openConnection().getInputStream()
-                )
-            )
-            //------------FOR DEMO -----------------
+                holder.avatarView.visibility = View.VISIBLE
+
+                // Display avatar if on feed
+                if (user?.profilePictureMetadata != null) {
 
-            // TODO : to change
-            holder.descriptionView.text = photo?.caption
+                    val avatarImage = db.getUserProfilePicture(user.profilePictureMetadata!!, userId)
+                    holder.avatarView.setImageBitmap(avatarImage)
+                }
+
+            } else {
+
+                // Do not display avatar if not on feed
+                holder.avatarView.visibility = View.GONE
+            }
+
+            // Display post content
+            if (photo is TrashPhotoMetadata) {
+
+                // Set default value
+                holder.titleView.text = user?.displayName ?: "Unknown user"
+                holder.subtitleView.text = (photo?.takenOn?.toString()
+                    ?: "Unknown date") + " | " + (photo?.trashCategory?.title
+                    ?: "No category")
+                holder.descriptionView.text = photo?.caption
+                holder.photoView.setImageBitmap(
+                    db.getImage(photo)
+                )
```

Comment:
Here it would be better if we had the strings in the String resource file. So that we can easily localize them.

There would be two option:

1. The best option to be able to achieve this, would be to separate the logic from the view, and delegate the logic to the caller of this Adapter, which in this case is `UserPhotoFragment`. So that we directly have the context. It would even allow you to call the `db` from outside the Adapter (Directly in the `UserPhotoFragment` class,) and therefor avoid to pass `lifecycleScope` as parameter to the adapter. You have an example of how I did it for the `EventsListAdapter` with the `EventDataAdapterListeners`. You would have to declare an interface, and then make the `UserPHotoFragment` implement this interface.
2. You can add the `context` as a parameter to the adapter, and directly access the string resources.

I personally find the first option cleaner, and it would also make it easier for you to test the Adapter I think, but the second option is probably easier.

Let me know what you think.

Reply:
Actually, it was already like that in the previous version. I tried to link it with the ressource file but I didn't manage to do it and I think this is the reason why it was left like that. I will try to implement what you suggested. Thank you very much.

Reply:
I took inspiration of what was already implemented in `UserViewAdapter` which correspond to the option 2. Thank you very much for your help